onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -2241,6 +2246,17 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsSetCustomJoinThreadFn, _Inout_ OrtSes
   API_IMPL_END
 }
 
+ORT_API(const OrtTrainingApi*, OrtApis::GetTrainingApi, uint32_t version) {
+#ifdef ENABLE_TRAINING_ON_DEVICE
+  return OrtTrainingApis::GetTrainingApi(version);
+#endif
+
+  ORT_UNUSED_PARAMETER(version);
+  fprintf(stderr, "Training APIs are not supported with this build. Please use an onnxruntime training build to retrieve the training APIs.\n");
```

Comment:
Given at the moment the only way to get these apis is when building from source, let's add this to the error message. Something like  - Please build onnxruntime trianing build from source with build flags enable_training and enable_training_on_device. Ideally, we should add some documentation for on device training and just add a link to it here.